misc/python/materialize/scalability/endpoint.py

```diff
@@ -42,3 +45,10 @@ def sql(self, sql: str) -> None:
         conn = self.sql_connection()
         cursor = conn.cursor()
         cursor.execute(sql)
+
+    def __str__(self) -> str:
```

Comment:
I consider it a bit surprising (for the caller) that the to-string methods opens a db connection and queries data to return its representation.
While some `get_version` method would sound more appropriate, feel free to keep it like this if simplifies things by allowing to handle all classes in the same fashion.

Reply:
You are right. I was lazy, will rework it .